### PR TITLE
fix(mesh): disables default network policy management

### PR DESCRIPTION
**Pull request by bartoszmajsak**
## Description

Disables automatic network policy management in Openshift Service Mesh. See https://docs.openshift.com/container-platform/4.14/service_mesh/v2x/ossm-traffic-manage.html#ossm-understanding-networkpolicy_traffic-management for more context.

Fixes https://issues.redhat.com/browse/RHOAIENG-1014

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: tests/integration/features/servicemesh_feature_test.go [hunk 1]
```diff
@@ -189,11 +266,63 @@ func createSMCPInCluster(cfg *rest.Config, smcpObj *unstructured.Unstructured, n
...
+
+	smcp, err := dynamicClient.Resource(gvr.SMCP).Namespace(namespace).Get(context.TODO(), name, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	return smcp, nil
+}
+
+func fromTestTmpDir(fileName string) string {
```
Comment (bartoszmajsak) [inline tests/integration/features/servicemesh_feature_test.go:291]: This is temporary and will be removed when https://github.com/opendatahub-io/opendatahub-operator/pull/797 gets merged

File: tests/integration/features/servicemesh_feature_test.go [hunk 2]
```diff
@@ -87,55 +49,170 @@ var _ = Describe("Service Mesh feature", func() {
...
+	Expect(envtest.WaitForCRDs(envTest.Config, []*apiextensionsv1.CustomResourceDefinition{smcpCrdObj}, crdOptions)).To(Succeed())
+
+	return smcpCrdObj
+}
+
+const serviceMeshControlPlaneCRD = `apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  labels:
+    maistra-version: 2.4.2
```
[the remark this answers is not in the document]
Comment (bartoszmajsak) [reply tests/integration/features/servicemesh_feature_test.go:178]: Doesn't matter for these tests I guess, or does it? At this point it's only CRD, there's no cluster behavior implied in the tests, as we do not install an actual operator.

[the remark this answers is not in the document]
Comment (zdtsw) [reply tests/integration/features/servicemesh_feature_test.go:178]: yeah, not matter for the test.
was only thinking to be as close as to the real env.

**Review by zdtsw — APPROVED** [review]
/lgtm

**Pull request merged**